jenkins_pipelines/environments/common/pipeline-build-validation-cleanup.groovy

```diff
@@ -0,0 +1,249 @@
+                if (params.targeted_project.contains("5.0")) {
+                    product_version = '5.0'
+                } else if (params.targeted_project.contains("4.3")) {
+                    product_version = '4.3'
+                } else if (params.targeted_project.contains("uyuni")) {
+                    product_version = 'uyuni'
+                }
+                else {
+                    // Use the `error` step instead of `throw`
+                    error("Error: targeted_project must contain either '5.0', '4.3' or uyuni.")
+                }
+            }
+
+            stage('Confirm Environment Cleanup') {
+                // Ask the user what environment they are cleaning, ensuring the answer matches params.targeted_project
+                def environmentChoice = input(
+                        message: 'What environment are you cleaning?',
+                        parameters: [
+                                string(name: 'Environment_Name', description: 'Enter the name of the environment you are cleaning.')
+                        ]
+                )
+
+                // Validate that the user entered the correct environment
+                if (environmentChoice != params.targeted_project) {
+                    error("The environment name entered does not match the targeted project. Aborting pipeline.")
+                }
+            }
+
+            stage("Copy terraform files from ${params.targeted_project}"){
+                // Copy tfstate and terraform directory to the result directory
+                sh """
+                    cp ${targetedTfStateFile} ${localTfStateFile}
+                    cp -r ${targetedTerraformDirPath}.terraform ${localSumaformDirPath}
+                """
+
+            }
+
+            stage("Extract the controller and server hostname") {
+                try {
+                    controllerHostname = sh(
+                            script: """
+                            set -e
+                            cd ${localSumaformDirPath}
+                            terraform output -json configuration | jq -r '.controller.hostname'
+                        """,
+                            returnStdout: true
+                    ).trim()
+
+                    serverHostname = sh(
+                            script: """
+                            set -e
+                            cd ${localSumaformDirPath}
+                            terraform output -json configuration | jq -r '.server.hostname'
+                        """,
+                            returnStdout: true
+                    ).trim()
+
+                    // Print the values for confirmation
+                    echo "Extracted controller hostname: ${controllerHostname}"
+                    echo "Extracted server hostname: ${serverHostname}"
+
+                } catch (Exception e) {
+                    error("Failed to extract hostnames: ${e.message}")
+                }
+            }
```

Comment:
I know it might be over engineering, but if you are parsing these values, you can also retrieve the `product_version`, that way we don't depend on the name of a Jenkins job to discover on which environment we are.

I'm thinking for instance in a cleanup of the Continuous Build Validation pipeline, that it gonna be parametrizable to run either Head, either 5.0

Reply:
Indeed this will create issues. I will not address this change in this PR but create a issue for that. We could use API or ssh command I suppose. https://github.com/SUSE/spacewalk/issues/26012